ngrinder-controller/src/main/java/org/ngrinder/perftest/service/PerfTestRunnable.java

```diff
@@ -403,6 +419,10 @@ void runTestOn(final PerfTest perfTest, GrinderProperties grinderProperties, fin
 
 		singleConsole.addListener(stopReason -> {
 			PerfTest fetchedPerftest = perfTestService.getOne(perfTest.getId());
+			if (isValidStopReason(stopReason)) {
```

Comment:
Is this code not necessary move to within `if (fetchedPerftest.getStatus().isStoppable()) {` block?

Reply:
check it only if it's stoppable, i agree!!